junit-vintage-engine/src/main/java/org/junit/vintage/engine/execution/TestRun.java

```diff
@@ -95,7 +95,12 @@ boolean isDescendantOfRunnerTestDescriptor(TestDescriptor testDescriptor) {
 		}
 		// @formatter:off
 		return descriptors.stream()
-				.filter(testDescriptor -> description == testDescriptor.getDescription())
+				.filter(testDescriptor -> {
+					// @marcphillip: why are Descriptions equated by reference and not using .equals()?
+					@SuppressWarnings("ReferenceEquality")
+					boolean result = (description == testDescriptor.getDescription());
+					return result;
+				})
```

Comment:
I should probably have added a comment to that particular piece of code. We use reference equality because `Description` are not necessarily equal even though their `equals` method returns true. For instance, it's quite simple to write a parameterized test class that produces different `Description` instances for a test method that are "equal" but not by reference.